optuna/samplers/_qmc.py

```diff
@@ -0,0 +1,168 @@
+from collections import OrderedDict
+from typing import Any
+from typing import Dict
+from typing import Optional
+
+import scipy
+
+import optuna
+from optuna import distributions
+from optuna import logging
+from optuna._transform import _SearchSpaceTransform
+from optuna.distributions import BaseDistribution
+from optuna.samplers import BaseSampler
+from optuna.study import Study
+from optuna.trial import FrozenTrial
+from optuna.trial import TrialState
+
+
+_logger = logging.get_logger(__name__)
+
+_SUGGESTED_STATES = (TrialState.COMPLETE, TrialState.PRUNED)
+
+_NUMERICAL_DISTRIBUTIONS = (
+    distributions.UniformDistribution,
+    distributions.LogUniformDistribution,
+    distributions.DiscreteUniformDistribution,
+    distributions.IntUniformDistribution,
+    distributions.IntLogUniformDistribution,
+)
+
+
+class QMCSampler(BaseSampler):
+    def __init__(
+        self,
+        *,
+        seed: Optional[int] = None,
+        qmc_type: str = "sobol",
+        independent_sampler: Optional[BaseSampler] = None,
+        search_space: Optional[Dict[str, BaseDistribution]] = None,
+        warn_independent_sampling: bool = True,
+    ) -> None:
+
+        self._seed = seed
+        self._independent_sampler = independent_sampler or optuna.samplers.RandomSampler(seed=seed)
+        self._qmc_type = qmc_type
+        self._qmc_engine = None
+        # TODO(kstoneriv3): make sure that search_space is either None or valid search space.
+        self._initial_search_space = search_space
```

Comment:
```suggestion
        self._fixed_search_space = search_space
```
Does it make it clearer?